src/track/beats.h

```diff
@@ -54,124 +84,264 @@ class Beats : public QObject {
+    /// while a strong reference to the Beats object to ensure that the Beats
+    /// object is not deleted. Caller takes ownership of the returned BeatsIterator
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual std::unique_ptr<BeatIterator> findBeats(double startSampleNum,
+            double stopSampleNum) const {
+        return findBeatsNew(startSampleNum / 2.0, stopSampleNum / 2.0);
+    }
+    virtual std::unique_ptr<BeatIterator> findBeatsNew(FrameNum startFrameNum,
+            FrameNum stopFrameNum) const;
+
+    /// Return whether or not a Beat lies between startFrameNum and endFrameNum
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual bool hasBeatInRange(double startSampleNum,
+            double stopSampleNum) const {
+        return hasBeatInRangeNew(startSampleNum / 2.0, stopSampleNum / 2.0);
+    }
+    virtual bool hasBeatInRangeNew(FrameNum startFrameNum,
+            FrameNum stopFrameNum) const;
+
+    /// Return the average BPM over the entire track if the BPM is
+    /// valid, otherwise returns -1
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual double getBpm() const {
+        return getBpmNew().getValue();
+    }
+    virtual Bpm getBpmNew() const;
+
+    /// Return the average BPM over the range from startFrameNum to endFrameNum,
+    /// specified in frames if the BPM is valid, otherwise returns -1
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual double getBpmRange(double startSampleNum,
+            FrameNum stopSampleNum) const {
+        return getBpmRangeNew(startSampleNum / 2.0, stopSampleNum / 2.0);
+    }
+    virtual double getBpmRangeNew(FrameNum startFrameNum,
+            FrameNum stopFrameNum) const;
+
+    /// Return the average BPM over the range of n*2 beats centered around
+    /// curFrameNum.  (An n of 4 results in an averaging of 8 beats).  Invalid
+    /// BPM returns -1.
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual double getBpmAroundPosition(double curSampleNum, int n) const {
+        return getBpmAroundPositionNew(curSampleNum / 2.0, n).getValue();
+    }
+    virtual Bpm getBpmAroundPositionNew(FrameNum curFrameNum, int n) const;
 
     virtual double getMaxBpm() const {
+        // TODO(JVC) Why is returning a constant? MaxBpm must be taken from somewhere
+        constexpr double kMaxBpm = 500;
         return kMaxBpm;
     }
 
-    ////////////////////////////////////////////////////////////////////////////
-    // Beat mutations
-    ////////////////////////////////////////////////////////////////////////////
+    /// Sets the track signature at the nearest frame
+    // TODO(JVC) Temporary adaptor. Will be removed before finalizing the PR
+    virtual void setSignature(TimeSignature signature, double sample = 0) {
```

Comment:
This function is for a new feature, so there is no point adding an adapter for the legacy way of specifying positions as a `double` for a sample position.